public/app/percona/dbaas/components/DBCluster/DBCluster.test.tsx

```diff
@@ -181,59 +199,61 @@ describe('DBCluster::', () => {
   });
 
   it('should open AddModal if kubernetesCluster was selected', async () => {
-    render(
-      <Provider
-        store={configureStore({
-          percona: {
-            user: { isAuthorized: true },
-            settings: { loading: false, result: { isConnectedToPortal: true, dbaasEnabled: true } },
-            kubernetes: {
-              loading: false,
-            },
-            dbaas: {
-              selectedKubernetesCluster: {
-                kubernetesClusterName: 'testPreselectedCluster',
-                operators: {
-                  psmdb: {
-                    availableVersion: '1.12.0',
-                    status: KubernetesOperatorStatus.ok,
-                    version: '1.11.0',
-                  },
-                  pxc: {
-                    availableVersion: undefined,
-                    status: KubernetesOperatorStatus.ok,
-                    version: '1.11.0',
+    await waitFor(() =>
```

Comment:
Why is the `waitFor` need here?

Should it be on the expect call?

Reply:
If I removed waitFor from render it returns me such error
![image](https://user-images.githubusercontent.com/90199600/197764697-bd43958f-9d3d-44d7-a20a-7cf0bf978b8b.png)

Reply:
Yeah that is weird, but is the waitFor really the solution here, or does it only mask a different error? Lets ask in slack, I think someone had a similiar error to this.

Reply:
Usually, `waitFor` is not needed in the render itself. That error comes from the fact that the page is not 100% ready when we test it (i.e. some state update happens and such). Whenever I write these tests, I use `waitFor` with an element I'm sure will be there after everything is ready. A good example is the loading spinner: there are many that `waitFor` the spinner to be removed before doing assertions.

Reply:
It's easier to understand RTL and how it works if we think of it from a user perspective: when is the page actually ready? Spinner missing, data on the table, and so forth.